examples/preprocessing/plot_movement_compensation.py

```diff
@@ -0,0 +1,111 @@
+rots = [x_rot, y_rot, z_rot, xz_rot, xmz_rot, yz_rot, mymz_rot]
+# The transpose of a rotation matrix is a rotation in the opposite direction
+rots += [rot.T for rot in rots]
+
+raw = mne.io.Raw(raw_fname).crop(0, len(rots))
+raw.load_data().pick_types(meg=True, stim=True, copy=False)
+raw.add_proj([], remove_existing=True)
+center = (0., 0., 0.04)  # a bit lower than device origin
+raw.info['dev_head_t']['trans'] = np.eye(4)
+raw.info['dev_head_t']['trans'][:3, 3] = center
+pos = np.zeros((len(rots), 10))
+for ii in range(len(pos)):
+    pos[ii] = np.concatenate([[ii], rot_to_quat(rots[ii]), center, [0] * 3])
+pos[:, 0] += raw.first_samp / raw.info['sfreq']  # initial offset
+
+# Let's activate a vertices bilateral auditory cortices
+src = mne.read_source_spaces(src_fname)
+labels = mne.read_labels_from_annot('sample', 'aparc.a2009s', 'both',
+                                    regexp='G_temp_sup-Plan_tempo',
+                                    subjects_dir=subjects_dir)
+assert len(labels) == 2  # one left, one right
+vertices = [np.intersect1d(l.vertices, s['vertno'])
+            for l, s in zip(labels, src)]
+data = np.zeros([sum(len(v) for v in vertices), int(raw.info['sfreq'])])
+activation = np.hanning(int(raw.info['sfreq'] * 0.2)) * 1e-9  # nAm
+t_offset = int(np.ceil(0.2 * raw.info['sfreq']))  # 200 ms in (after baseline)
+data[:, t_offset:t_offset + len(activation)] = activation
+stc = mne.SourceEstimate(data, vertices, tmin=-0.2,
+                         tstep=1. / raw.info['sfreq'])
+
+# Simulate the movement
+raw = simulate_raw(raw, stc, trans_fname, src, bem_fname,
+                   head_pos=pos, interp='zero', n_jobs=-1)
+raw_stat = simulate_raw(raw, stc, trans_fname, src, bem_fname,
+                        head_pos=None, n_jobs=-1)
+
+##############################################################################
+# Process our simulated raw data taking into account head movements
+
+# extract our resulting events
+events = mne.find_events(raw, stim_channel='STI 014')
+assert len(events) == len(pos)  # make sure we did this right
+assert np.array_equal(events[:, 2], np.arange(len(pos)) + 1)
+events[:, 2] = 1
+raw.plot(events=events)
+
+topo_kwargs = dict(times=[0, 0.1, 0.2], ch_type='mag', vmin=-500, vmax=500)
+
+# 0. Take average of stationary data (bilateral auditory patterns)
+evoked_stat = mne.Epochs(raw_stat, events, 1, -0.2, 0.8).average()
+evoked_stat.plot_topomap(title='Stationary', **topo_kwargs)
+
+# 1. Take a naive average (smears activity)
+evoked = mne.Epochs(raw, events, 1, -0.2, 0.8).average()
+assert evoked.nave == len(pos)
+evoked.plot_topomap(title='Moving: naive average', **topo_kwargs)
+
+# 2. Use raw movement compensation (works well)
+raw_sss = maxwell_filter(raw, pos=pos)
+evoked_raw_mc = mne.Epochs(raw_sss, events, 1, -0.2, 0.8).average()
+assert evoked_raw_mc.nave == len(pos)
+evoked_raw_mc.plot_topomap(title='Moving: movement compensated', **topo_kwargs)
```

Comment:
Actually thinking about it a bit more, `sample` is a nicely self-contained, mostly unchanging dataset. It's also big so I feel bad making people re-download it. Is it okay if I make a new one called `MNE-misc-data`? Hopefully it could stay a bit smaller as we add miscellaneous things to it.